scimono-server/src/main/java/com/sap/scimono/api/Groups.java

```diff
@@ -85,6 +83,16 @@ public Groups(@Context Application appContext, @Context UriInfo uriInfo) {
     groupPreProcessor = ResourcePreProcessor.forGroups(resourceLocationService, groupAPI, resourceTypesAPI, schemaAPI);
   }
 
+  public Groups(SCIMApplication scimApplication, @Context UriInfo uriInfo) {
+
+    groupAPI = scimApplication.getGroupsCallback();
+    schemaAPI = scimApplication.getSchemasCallback();
+    resourceTypesAPI = scimApplication.getResourceTypesCallback();
+    scimConfig = scimApplication.getConfigurationCallback();
+    resourceLocationService = new ResourceLocationService(uriInfo, scimConfig, GROUPS);
+    groupPreProcessor = ResourcePreProcessor.forGroups(resourceLocationService, groupAPI, resourceTypesAPI, schemaAPI);
+  }
```

Comment:
Please remove this constructor and use a SCIMApplication stub for testing.

Reply:
Done.